cache/cache.go

```diff
@@ -57,3 +60,34 @@ func (c *Cache) WriteTo(w io.Writer) (int64, error) {
 	}
 	return total, nil
 }
+
+// Index generates an index for the given cache on a particular
+// column.  This is required for caches beyond a libnss-cache defined
+// size in order for them to be read correctly.
+func (c *Cache) Index(col int) bytes.Buffer {
+	var ordered []string
+	mapped := make(map[string]Entry, len(c.entries))
+	for i := range c.entries {
+		ordered = append(ordered, c.entries[i].Column(col))
+		mapped[c.entries[i].Column(col)] = c.entries[i]
+	}
+
+	// libnss-cache depends on the indexes being ordered in order
+	// to accelerate the system with a binary search.
+	sort.Strings(ordered)
+
+	var b bytes.Buffer
+	var offset int64
+	for i := range ordered {
```

Comment:
Wouldn't `for i, key := range ordered {` give you the expected key without having to access `ordered` again?

Reply:
Yeah this whole loop is a bit of a mess, I've refactored it to clean up the number of times lookups needed to happen.